### PR TITLE
[SDESK-6199] Ability to search all wire versions

### PR DESCRIPTION
**Pull request by MarkLark86**
Depends on https://github.com/superdesk/superdesk-core/pull/2282

### CODE REVIEW

File: data_updates/00003_20211011-162134_items.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class DataUpdate(_DataUpdate):
+    """Adds ``original_id`` mapping to the ``items`` resource
+
+    This is done in a ``DataUpdate`` so a complete re-index is not required
+    As this is a new field and not used in previous versions, we can safely add the mapping
+    Otherwise a re-index will require a longer down-time when upgrading NewsHub
+
+    We're also adding this mapping manually otherwise Elasticsearch will
+    set it as text with keyword sub mapping.
```
Comment (petrjasek) [inline data_updates/00003_20211011-162134_items.py:22]: wouldn't it actually get the new field via initialize_data command?

File: newsroom/search.py [hunk 2]
```diff
@@ -67,14 +67,97 @@ class BaseSearchService(Service):
...
+        internal_req = self.get_internal_request(search)
+        earlier_versions = self.internal_get(internal_req, search.lookup)
+        next_item_ids = [
+            str(self.get_last_version(doc)['_id'])
+            for doc in earlier_versions.docs
+        ]
+
+        # Now run a query only using the IDs from the above search
+        # This final search makes sure pagination still works
+        search.query['bool']['must_not'] = []
```
Comment (petrjasek) [inline newsroom/search.py:106]: I guess you could just override the bool part:
```
search.query['bool'] = {'must': {'terms': ...}}

**Review by petrjasek — APPROVED** [review]

File: newsroom/search.py [hunk 3]
```diff
@@ -66,17 +66,112 @@ class BaseSearchService(Service):
...
+            req = ParsedRequest()
+            req.args = {
+                'source': json.dumps({
+                    'query': {
+                        'bool': {
+                            'must': [
+                                {'term': {'original_id': original_id}},
+                            ],
+                            'must_not': [
+                                {'constant_score': {'filter': {'exists': {'field': 'nextversion'}}}}
```
Comment (petrjasek) [inline newsroom/search.py:146]: just wondering why using the constant score here?

Comment (MarkLark86) [reply newsroom/search.py]: Just bad habit I suppose from Elastic 1.x days

**Pull request merged**